### DISCUSSION TITLE
Error: Call to a member function addMedia() on null

### DISCUSSION THREAD

**Discussion by mattgreenfield** [spatie/laravel-medialibrary#2739] [Q&A]
Hi,
I'm fairly new to laravel and can't get my head around this error: `Call to a member function addMedia() on null`.

It's coming from my controller here:
```
class MarketingSiteController extends Controller
{
    public function index(): View
    {
        $media = MarketingImages::find(1)->addMedia(resource_path('marketing/_assets/images/blah.png'))->toMediaCollection('marketing');
        return view('layout', ['image' => $media]);

```
I see that the `find` isn't finding anything but what should it be finding? This is me adding the image to the collection right?

My `marketing_images` table exists in my DB but is empty.

My `MarketingImage` Model looks like this 
```
<?php

declare(strict_types=1);

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Spatie\MediaLibrary\HasMedia;
use Spatie\MediaLibrary\InteractsWithMedia;

class MarketingImages extends Model implements HasMedia
{
    use InteractsWithMedia;
}
```

Any help would be much appreciated!

As an aside, all I want to do is to optimise and make multiple sizes for some images in a folder in my code base. Is this library overkill? It talks a lot about image uploads.

Thanks for your help,
Matt

**Comment by erikn69** [comment]
>I see that the find isn't finding anything but what should it be finding?
>My marketing_images table exists in my DB but is empty.

 if it is empty, why do you search for `1`?

**Reply by mattgreenfield** [reply]
I was following the guide here https://spatie.be/docs/laravel-medialibrary/v8/working-with-media-collections/simple-media-collections
My assumption was that `->addMedia()` would take a file from my source directory of images and put it in to the collection so I can use it elsewhere (optimised and resized).
If that's not the case, how do I actually add an image to that database table?

thanks

**Reply by mattgreenfield** [reply]
@PaolaRuby @erikn69 - thanks for the laughing emoji but any actual help would be great please.

**Reply by fredsal** [reply]
>I see that the find isn't finding anything but what should it be finding? 

Maybe try to learn laravel first and then you will find the problem
https://laravel.com/docs/8.x/eloquent#retrieving-models
https://laravel.com/docs/8.x/eloquent-relationships#polymorphic-relationships
